.github/workflows/build-package-docs.yaml

```diff
@@ -108,10 +108,16 @@ jobs:
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
 
-    - name: Log the workflow inputs if deploy checked
-      if: fromJSON(github.event.inputs.deploy)
+  check-deploy:
```

Comment:
Should this job only be run when `github.event_name == 'workflow_dispatch' && github.event.inputs.deploy == 'true'`?

Reply:
Also, I think this should have a `needs:` on `build-package-docs`.

Reply:
> Should this job only be run when `github.event_name == 'workflow_dispatch' && github.event.inputs.deploy == 'true'`?

Yes, those are the two conditions for the deploy job. As a follow up to this PR we can add a schedule to the workflow. I guess that's not clear right now but the triggers will be `workflow_dispatch` and `schedule`. We only want to run the deploy job for manually triggered runs.

Reply:
> > Should this job only be run when `github.event_name == 'workflow_dispatch' && github.event.inputs.deploy == 'true'`?
> 
> Yes, those are the two conditions for the deploy job.

Ah, it looks like the `if:` is there on the [task level](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif). I would move it to the [job level](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idif) so the job isn't created and provisioned just to do nothing. Than you can rely on `needs` to entirely skip the next job instead of having to duplicate the same `if` twice.

Reply:
> Also, I think this should have a `needs:` on `build-package-docs`.

Looking at this again and I don't think the `check-deploy` job has a dependency on `build-package-docs`. It simply checks if conditions are met to deploy the build and can run concurrently to the build job. Perhaps adding `needs` would make the jobs run sequentially but I'm not sure that is necessary.

I'm open to the suggestion anyway. I'd just like to make sure I understand why `needs` might be good here.

Reply:
Great suggestion @gotmax23 - I've pushed an update. Nice to get rid of the duplicated `if` which I didn't really like myself.

Reply:
> I'm open to the suggestion anyway. I'd just like to make sure I understand why needs might be good here.

I don't think it makes sense to print the deployment information if nothing will happen because the docs build failed.